### PR TITLE
Allow manual threshold checks

### PR DESCRIPTION
**Pull request by ansermino**
## Changes
- Adds callable function to allow any user to execute a proposal that meets the current threshold
- Separates vote commit from status update

### CODE REVIEW

File: chainbridge/src/lib.rs [hunk 1]
```diff
@@ -300,6 +306,25 @@ decl_module! {
...
+        /// # <weight>
+        /// - weight of proposed call, regardless of whether execution is performed
+        /// # </weight>
+        #[weight = FunctionOf(
+            |args: (&DepositNonce, &ChainId, &Box<<T as Trait>::Proposal>)| args.2.get_dispatch_info().weight + 500_000,
+            |args: (&DepositNonce, &ChainId, &Box<<T as Trait>::Proposal>)| args.2.get_dispatch_info().class,
+            true
+        )]
+        pub fn eval_vote_state(origin, nonce: DepositNonce, src_id: ChainId, prop: Box<<T as Trait>::Proposal>) -> DispatchResult {
+            ensure_signed(origin)?;
```
Comment (noot) [inline chainbridge/src/lib.rs:324]: where is this defined?

File: chainbridge/src/lib.rs [hunk 2]
```diff
@@ -300,6 +306,25 @@ decl_module! {
...
         }
+
+        /// Evaluate the state of a proposal given the current vote threshold.
+        ///
+        /// A proposal with enough votes will be either executed or cancelled, and the status
+        /// will be updated accordingly.
+        ///
+        /// # <weight>
+        /// - weight of proposed call, regardless of whether execution is performed
+        /// # </weight>
```
Comment (noot) [inline chainbridge/src/lib.rs:317]: is this for docs? idk how the formatting works

File: chainbridge/src/lib.rs (hunk 1 above)
Comment (ansermino) [reply chainbridge/src/lib.rs:324]: It's provided by frame_support: https://crates.parity.io/frame_system/fn.ensure_signed.html

File: chainbridge/src/lib.rs (hunk 2 above)
Comment (ansermino) [reply chainbridge/src/lib.rs:317]: Yeah, seems to be the convention in other palettes. Doesn't seem to do much in the rust docs but I think they probably parse it out for the runtime metadata, so something like a UI can use it.

Eg usage: https://crates.parity.io/pallet_democracy/struct.Module.html#method.delegate

**Pull request merged**